### PR TITLE
Changing Lambda Integration Type to AWS_PROXY

### PR DESCRIPTION
**Pull request by fishcharlie**
According to the [AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html) integration type should be `AWS_PROXY` for Lambda functions. This really tripped me up for a while since it seems like the example is just wrong here.

Previously was getting the following error when trying to use `AWS` like the example shows:

> Error: error creating API Gateway v2 integration: BadRequestException: Currently, an API with a protocol type of HTTP may only be associated with proxy integrations (AWS_PROXY, HTTP_PROXY)

### CODE REVIEW

**Review by fishcharlie — COMMENTED** [review]
Changing the integration type to AWS_MOCK.

File: website/docs/r/apigatewayv2_integration.html.markdown [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource "aws_lambda_function" "example" {
 
 resource "aws_apigatewayv2_integration" "example" {
   api_id           = aws_apigatewayv2_api.example.id
-  integration_type = "AWS"
+  integration_type = "AWS_MOCK"
```
Comment (justinretzolk) [inline website/docs/r/apigatewayv2_integration.html.markdown:38]: Unless I'm misunderstanding, based on the [document you linked in the PR](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html), it seems that this should instead be `AWS_PROXY`. It doesn't appear that `AWS_MOCK` is a valid integration type, and `MOCK` appears to be used only for testing. If I've misunderstood, I'd be happy to take another look! 🙂 

```suggestion
  integration_type = "AWS_PROXY"
```

**Review by justinretzolk — CHANGES_REQUESTED** [review]
Thank you for taking the time to raise this PR! I had one note that I'd like us to take a look at before we get this merged in.

**Comment by fishcharlie** [comment]
@justinretzolk You are correct. My bad. Just committed that change.

**Review by justinretzolk — APPROVED** [review]
Looks good to me! Thank you for your contribution!

**Pull request merged**